src/Utilities/Serialization/PupBoost.hpp

```diff
@@ -53,4 +54,23 @@ template <class T, size_t N>
 void operator|(PUP::er& p, boost::container::static_vector<T, N>& v) {
   pup(p, v);
 }
+
+template <typename T>
+void pup(PUP::er& p, boost::math::quaternion<T>& quaternion) {
+  T component_1 = quaternion.R_component_1();
+  T component_2 = quaternion.R_component_2();
+  T component_3 = quaternion.R_component_3();
+  T component_4 = quaternion.R_component_4();
+  p | component_1;
+  p | component_2;
+  p | component_3;
+  p | component_4;
+  quaternion = boost::math::quaternion<T>(component_1, component_2, component_3,
+                                          component_4);
```

Comment:
Should this be wrapped in `if (p.isUnpacking())`? Seems a bit odd to modify an existing object that you are just packing.